src/docfx/build/toc/BuildTableOfContents.cs

```diff
@@ -11,26 +11,33 @@ namespace Microsoft.Docs.Build
 {
     internal static class BuildTableOfContents
     {
-        public static (IEnumerable<Error>, TableOfContentsModel, DependencyMap) Build(
+        public static (IEnumerable<Error>, TableOfContentsModel, DependencyMap, List<string>) Build(
             Context context, Document file, TableOfContentsMap tocMap)
         {
             Debug.Assert(file.ContentType == ContentType.TableOfContents);
 
             if (!tocMap.Contains(file))
             {
-                return (Enumerable.Empty<Error>(), null, DependencyMap.Empty);
+                return (Enumerable.Empty<Error>(), null, DependencyMap.Empty, new List<string>());
             }
 
             // todo: build resource files linked by toc
             var dependencyMapBuilder = new DependencyMapBuilder();
-            var (errors, tocModel, tocMetadata, refArticles, refTocs) = Load(context, file, dependencyMapBuilder);
+            var (errors, tocModel, tocMetadata, refArticles, refTocs, monikers) = Load(context, file, dependencyMapBuilder, true);
```

Comment:
>monikers [](start = 70, length = 8)

Can we get `monikers` separately here? So that we do not need to pass in a flag to identify if we need to do it, also not pollute the other usages.

Reply:
we can remove this flag, bu the `monikers` is the collection of every node's monikers, it should be a return value

---
In reply to: [235288217](https://github.com/dotnet/docfx/pull/3655#discussion_r235288217) [](ancestors = 235288217)